File: services/ms-voo/src/main/java/airportsystem/msvoo/repository/VooRepository.java
package airportsystem.msvoo.repository;

import airportsystem.msvoo.model.Voo;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.time.LocalDateTime;
import java.util.List;

@Repository
public interface VooRepository extends JpaRepository<Voo, String> {

    Voo findByCodigo(String codigo);
    Voo findByOrigem(String origem);
    Voo findByDestino(String destino);

    List<Voo> findByDataHoraBetween(LocalDateTime dataInicio, LocalDateTime dataFim);

    boolean existsByCodigo(String codigo);
    boolean existsByOrigem(String origem);
    boolean existsByDestino(String destino);

}

File: services/ms-voo/src/main/kotlin/com/example/airport_system/repository/VooRepository.kt
package com.example.airport_system.ms_voo.repository

import com.example.airport_system.ms_voo.model.Voo
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface VooRepository : JpaRepository<Voo, String> {
    // Aqui você pode adicionar métodos personalizados, se necessário
    fun findByOrigem(origem: String): List<Voo>
    fun findByDestino(destino: String): List<Voo>
    fun findByDataHora(dataHora: String): List<Voo>

}
